packages/gatsby-theme-docs/src/components/release-notes-filter-dates.js

```diff
@@ -28,29 +44,38 @@ const ReleaseNotesFilterDates = () => {
       <SpacingsStack scale="xs">
         <DateLabel htmlFor="from-filter-date">From</DateLabel>
         <div>
-          <DateInput
+          <DateInputField
+            type="date"
             id="from-filter-date"
+            max={maximumDate}
             value={filterParams.fromFilterDate || ''}
+            onClick={handleClick}
```

Comment:
I added the `showPicker` function only to the first date input so that we can test it and decide if we want to have it or not. When the picker is open, there is no way to write the date directly into the field, you would have to select it via the picker (in chrome). If we don't use it, you would still be able to write it directly into the field (which is still a bit problematic because of the bug we mentioned above).
Preview:
https://docskit-pr-1446.commercetools.vercel.app/docs-smoke-test/releases

Reply:
> we have a bug with keyboard input in the year field. it works flawlessly in the barebones HTML spec example ( https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#value ) but in our implementation it takes only the first number types and then exits the field. We need to revisit the onChange handler maybe? I am a bit clueless. onInput fires even more often. Something about React state and rerendering maybe

I tried to find out what the problem is and even without the `onChange` handler, the input field gets submitted (or reseted) every time you reach the year field. I also looked into the ui-kit component and I noticed that the behaviour is slightly different. It seems like the date is only being submitted after pressing enter.

@emmenko maybe you have an idea how we can solve this?

Reply:
> even without the onChange handler, the input field gets submitted (or reseted) every time you reach the year field. 
That's really weird given that the plain HTML examples in the HTML reference docs don't do that.  Does React or some library we use magically attach some kind of even handlers or does some parent component of this input form trigger a re-render in certain situations?

Reply:
@timonrey 
This strange behaviour definitely has to do with the fact that the field is "controlled"meaning that the value is set by `filterParams.fromFilterDate` which is in turn changed by the handleOnFromFilterDateChange. Probably in some case the date is either not valid or it causes the render in the field to become weird and the field starts misbehaving. 
Try to comment out `value` and `onChange` props, and you'll see it working as expected.
One solution could be to call `setFilterParams` only when the date has a valid date value.

Let me know if you need further support on this

Reply:
> One solution could be to call setFilterParams only when the date has a valid date value.

That is a good idea in any case because there should never be situations where are shareable URL contains an invalid date.  But be aware that it's not exactly trivial to "validate".   E.g.  the case that my UI stuck on was  0020-02-21 ,   that is a valid date.  It's roundabout where christ was born but formally a valid date.  

maybe you need to introduce component level state ( useState // useEffect based) and not only rely on the URL being the UI state.   If the state "loop" is managed inside react it the "controlled" aspect mabye behaves more predictable because it does not loop through the browser engine any more but react has full control?

Reply:
Here's an example (ignore the use of `moment`)
The idea is: you apply your logic, if the date is valid, you format it and you pass it as value to the input
https://codesandbox.io/s/funny-chebyshev-nf9dv?file=/src/App.js